azure/services/roleassignments/roleassignments.go

```diff
@@ -94,7 +107,8 @@ func (s *Service) reconcileVM(ctx context.Context, roleSpec azure.RoleAssignment
 		return errors.Errorf("%T is not a compute.VirtualMachine", resultVMIface)
 	}
 
-	err = s.assignRole(ctx, roleSpec.Name, resultVM.Identity.PrincipalID)
+	roleSpec.PrincipalID = resultVM.Identity.PrincipalID
```

Comment:
It looks like this is the only spot where we can't get a spec value from `machine.go`. If we can pass this value in as an argument for `RoleAssignmentSpecs()` then that we could avoid the need to cast the spec type directly. Let me see if I can figure something out.